gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/AccountFactorsEndpointHandler.java

```diff
@@ -290,12 +299,15 @@ public void verifyFactor(RoutingContext routingContext) {
                 final EnrolledFactor enrolledFactor = optionalEnrolledFactor.get();
                 verifyFactor(code, enrolledFactor, factorProvider, vh -> {
                     if (vh.failed()) {
+                        updateAuditLog(MFA_ENROLL_FAILED, user, client, h.result(), enrolledFactor, vh.cause());
                         routingContext.fail(vh.cause());
                         return;
                     }
 
                     // verify successful, change the EnrolledFactor status and increment moving factor
                     enrolledFactor.setStatus(FactorStatus.ACTIVATED);
+                    updateAuditLog(MFA_ENROLL_SUCCEED, user, client, h.result(), enrolledFactor, null);
+
```

Comment:
missing log in line 320 - catch block?

Reply:
There is an error handler attached on the AccountProvider#175 such as `accountRouter.route().failureHandler(new ErrorHandler());`
Any routingContext fail in this class error is logged by the ErrorHandler.java class